zerver/actions/message_edit.py

```diff
@@ -1356,6 +1378,20 @@ def get_message_update_data(
     return content, rendering_result, prior_mention_user_ids, mention_data, links_for_embed
 
 
+def check_resolve_topic_permissions(
+    user_profile: UserProfile,
+    message: Message,
+    stream_id: Optional[int],
+) -> None:
+    if not user_profile.can_resolve_topic():
+        raise JsonableError(_("You don't have permission to resolve/unresolve topics."))
+
+    if stream_id is None:
+        stream_id = message.recipient.type_id
+        stream = get_stream_by_id_in_realm(stream_id, user_profile.realm)
+        check_stream_access_based_on_stream_post_policy(user_profile, stream)
```

Comment:
I am not sure whether we want to check the permission is old stream even when the topic is being moved to new stream.

Reply:
`stream_id` is None here indicates that message is not being moved to new stream. So, `stream_post_policy` of current stream is being checked when the topic is resolved in this case.

Reply:
I was thinking whether we need to check posting permission in both old and new streams. Anyways, I just started a discussion on https://chat.zulip.org/#narrow/stream/101-design/topic/managing.20resolve.20topic.20permissions.20independently.20.2321811 for clarifying the behavior with respect to posting permissions.

Reply:
thanks! I should have posted there earlier -- would've avoided avoided the confusion.